### PR TITLE
Refactor(eos_designs)!: MPLS Peer Logic

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Breaking change for MPLS Beta feature!

Refactor(eos_designs): MPLS Peer Logic

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Remove default rr<->rr peerings. Instead we will leverage `mpls_route_reflectors` on the `route_reflectors` themselves - similar to what we do with EVPN. This change allows for more flexible RR designs, including hierarchical RRs.
- Align MPLS client/server logic with evpn templates.
- Update `avd_overlay_peers` to also cover mpls route_reflectors
- Leverage `avd_overlay_peers` fact instead of accessing group vars directly (hardcoded `rr`)

## How to test

- Tested with existing molecule scenarios.
- Added the now required `mpls_route_reflectors` to the RRs, to avoid changes.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by emilarista — APPROVED** [review]
Have tested with my repo as well. LGTM!

File: ansible_collections/arista/avd/roles/eos_designs/templates/overlay/logic-mpls-route-servers.j2 [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
 {%             set overlay_pe_vars = avd_switch_facts[fabric_switch] %}
 {%             if overlay_pe_vars.switch.mpls_overlay_role is arista.avd.defined('client') %}
 {%                 set peer_pe = namespace() %}
 {%                 set peer_pe.bgp_as = overlay_pe_vars.switch.bgp_as %}
-{%                 set peer_pe.ip_address = overlay_pe_vars.switch.router_id %}
+{%                 if switch.overlay_routing_protocol_address_family is arista.avd.defined("ipv6") %}
+{%                     set peer_pe.ip_address = overlay_pe_vars.switch.ipv6_router_id %}
+{%                 else %}
+{%                     set peer_pe.ip_address = overlay_pe_vars.switch.router_id %}
+{%                 endif %}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/templates/overlay/logic-mpls-route-servers.j2]: We repeat this block 3 times in the PR - would it be worth making some facts out of this? like `switch.overlay_router_id` that would automatically return the appropriate router id depending on the `overlay_routing_protocol_address_family`?

**Review by gmuloc — APPROVED** [review]
LGTM as it is - only one question regarding a repeating pattern in Jinja (cf comment)

File: ansible_collections/arista/avd/roles/eos_designs/templates/overlay/logic-mpls-route-servers.j2 (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/overlay/logic-mpls-route-servers.j2]: Yeah I think that is a good solution. Let me add that to this PR.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/overlay/logic-mpls-route-servers.j2]: fixed

**Pull request merged**